drizzlepac/haputils/svm_quality_analysis.py

```diff
@@ -270,12 +272,11 @@ def compare_num_sources(catalog_list, drizzle_list, json_timestamp=None, json_ti
         # was an error.  However, for the purposes of this program, it is OK
         # that no catalog was produced.
         for catalog in cat_names:
-            does_exist = any(catalog in x for x in catalog_list)
+            if catalog in catalog_files:
 
-            # if the catalog exists, open it and find the number of sources string
-            num_sources = -1
-            cat_type = ""
-            if does_exist:
+                # if the catalog exists, open it and find the number of sources string
+                num_sources = -1
+                cat_type = ""
                 file = open(catalog, 'r')
```

Comment:
For future reference, the best practice for working with file I/O is to use 'with open(catalog, 'r') as file'  and indent all the code related to 'file'.  This will automatically close the file handle upon completion of the code indented code.